src/NzbDrone.Core/DecisionEngine/QualityUpgradableSpecification.cs

```diff
@@ -6,8 +7,8 @@ namespace NzbDrone.Core.DecisionEngine
 {
     public interface IQualityUpgradableSpecification
```

Comment:
I see the separation a bit hard. As I see it you have 3 scenarios:
1. You want best quality and once a language is donwloaded you want to stick with it (current situation).
2. You want best quality and doesn't care about language given a list (new situation, you can upgrade language if quality doesn't get downgrade).
3. You want best language possible, even if that mean you downgrade quality (new scenario)

I think the three are linked to the "quality" of the file downloaded, since the language is also a quality measure, at least IMO.